src/account_manager.rs

```diff
@@ -456,12 +457,16 @@ async fn poll(storage_path: PathBuf, syncing: bool) -> crate::Result<()> {
 async fn discover_accounts(
     storage_path: &PathBuf,
     client_options: &ClientOptions,
+    signer_type: Option<SignerType>,
 ) -> crate::Result<Vec<SyncedAccount>> {
     let mut synced_accounts = vec![];
     loop {
-        let mut account = AccountInitialiser::new(client_options.clone(), &storage_path)
-            .skip_persistance()
-            .initialise()?;
+        let mut account_initialiser =
+            AccountInitialiser::new(client_options.clone(), &storage_path).skip_persistance();
```

Comment:
seems like a good scenario for the `Builder` pattern here. So instead of `AccountInitialiser`, I'ld introduce `AccountBuilder`, that has a `finish` method, that returns an `Account`. Part of this pattern is also to have a `build` method on `Account`, that returns the `AccountBuilder`. In user code this `AccountBuilder` doesn't have to be explicitly imported.

Reply:
yeah the initialiser is something like a builder, but only accessible through the AccountManager

Reply:
you'ld for example use it like this:
`let account = Account::build(client_options.clone(), &storage_path)
    .skip_persistence()
    .initialise()?
    .finish();`